src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -1110,6 +1201,12 @@ void Instance::OnFailSafeTimerExpired()
     ChipLogDetail(Zcl, "Failsafe timeout, tell platform driver to revert network credentials.");
     mpWirelessDriver->RevertConfiguration();
     mAsyncCommandHandle.Release();
+
+    // Reset state on failsafe expiry.
```

Comment:
I don't see a spec basis for this...

Furthermore, if the failsafe was for something totally unrelated to commissioning or networks (e.g. an UpdateNOC), clearing this state doesn't obviously seem like the right thing to do.